[2025-09-15T06:50:58.242+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: spotify_saved_tracks_etl_pipeline.gold_saved_transform manual__2025-09-15T06:34:35.208642+00:00 [queued]>
[2025-09-15T06:50:58.251+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: spotify_saved_tracks_etl_pipeline.gold_saved_transform manual__2025-09-15T06:34:35.208642+00:00 [queued]>
[2025-09-15T06:50:58.251+0000] {taskinstance.py:1361} INFO - Starting attempt 4 of 6
[2025-09-15T06:50:58.349+0000] {taskinstance.py:1382} INFO - Executing <Task(PythonOperator): gold_saved_transform> on 2025-09-15 06:34:35.208642+00:00
[2025-09-15T06:50:58.359+0000] {standard_task_runner.py:57} INFO - Started process 32278 to run task
[2025-09-15T06:50:58.363+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'spotify_saved_tracks_etl_pipeline', 'gold_saved_transform', 'manual__2025-09-15T06:34:35.208642+00:00', '--job-id', '823', '--raw', '--subdir', 'DAGS_FOLDER/spotify_saved_dag.py', '--cfg-path', '/tmp/tmp8h45kpmo']
[2025-09-15T06:50:58.367+0000] {standard_task_runner.py:85} INFO - Job 823: Subtask gold_saved_transform
[2025-09-15T06:50:58.430+0000] {task_command.py:415} INFO - Running <TaskInstance: spotify_saved_tracks_etl_pipeline.gold_saved_transform manual__2025-09-15T06:34:35.208642+00:00 [running]> on host 0d0cd054d690
[2025-09-15T06:50:58.501+0000] {taskinstance.py:1660} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='spotify_saved_tracks_etl_pipeline' AIRFLOW_CTX_TASK_ID='gold_saved_transform' AIRFLOW_CTX_EXECUTION_DATE='2025-09-15T06:34:35.208642+00:00' AIRFLOW_CTX_TRY_NUMBER='4' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-09-15T06:34:35.208642+00:00'
[2025-09-15T06:50:58.519+0000] {logging_mixin.py:151} INFO - Executed sql/gold/saved_tracks/fact_saved_songs.sql
[2025-09-15T06:50:58.529+0000] {taskinstance.py:1943} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 192, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 209, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/etl/db_utils.py", line 11, in run_sql_scripts
    cur.execute(sql)
psycopg2.errors.DatatypeMismatch: column "song_duration_sec" is of type numeric but expression is of type text
LINE 9:     artist_id,
            ^
HINT:  You will need to rewrite or cast the expression.

[2025-09-15T06:50:58.542+0000] {taskinstance.py:1400} INFO - Marking task as UP_FOR_RETRY. dag_id=spotify_saved_tracks_etl_pipeline, task_id=gold_saved_transform, execution_date=20250915T063435, start_date=20250915T065058, end_date=20250915T065058
[2025-09-15T06:50:58.556+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 823 for task gold_saved_transform (column "song_duration_sec" is of type numeric but expression is of type text
LINE 9:     artist_id,
            ^
HINT:  You will need to rewrite or cast the expression.
; 32278)
[2025-09-15T06:50:58.587+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2025-09-15T06:50:58.603+0000] {taskinstance.py:2784} INFO - 0 downstream tasks scheduled from follow-on schedule check
